docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile

```diff
@@ -181,3 +182,25 @@ function su_humsci_profile_simplify_condition_forms(array &$condition_elements,
     unset($condition_elements['node_type']);
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function hs_field_helpers_eck_entity_type_insert(EntityInterface $entity) {
+  $eck_type = $entity->id();
+  // When a new ECK entity type is create, set initial permissions so that
+  // site builders aren't required to search for the necessary permissions.
+  user_role_grant_permissions(RoleInterface::ANONYMOUS_ID, ["view any $eck_type entities"]);
+  user_role_grant_permissions(RoleInterface::AUTHENTICATED_ID, ["view any $eck_type entities"]);
+
+  user_role_grant_permissions('contributor', [
```

Comment:
Perhaps some validation on the role name and that it exists? Otherwise, this looks good to go.

Reply:
D8 is way better than D7. it does a check on the role for us 
```
function user_role_grant_permissions($rid, array $permissions = []) {
  // Grant new permissions for the role.
  if ($role = Role::load($rid)) {
    foreach ($permissions as $permission) {
      $role->grantPermission($permission);
    }
    $role->trustData()->save();
  }
}
```

Reply:
just glad its not a random numeric rid like on D7.... ugh. that was annoying